utils/src/metrics.rs

```diff
@@ -109,46 +121,14 @@ lazy_static! {
         Arc::new(Mutex::new(ErrorHolder::new(500, 50 * 1024)));
 }
 
-#[derive(Default, Debug, Serialize)]
-pub struct GlobalIoStats {
-    // Whether to enable each file accounting switch.
-    // As fop accounting might consume much memory space, it is disabled by default.
-    // But global fop accounting is always working within each Rafs.
-    files_account_enabled: AtomicBool,
-    access_pattern_enabled: AtomicBool,
-    record_latest_read_files_enabled: AtomicBool,
-    // Given the fact that we don't have to measure latency all the time,
-    // use this to turn it off.
-    measure_latency: AtomicBool,
-    id: String,
-    // Total bytes read against the filesystem.
-    data_read: BasicMetric,
-    // Cumulative bytes for different block size.
-    block_count_read: [BasicMetric; BLOCK_READ_SIZES_MAX],
-    // Counters for successful various file operations.
-    fop_hits: [BasicMetric; StatsFop::Max as usize],
-    // Counters for failed file operations.
-    fop_errors: [BasicMetric; StatsFop::Max as usize],
-    // Cumulative latency's life cycle is equivalent to Rafs, unlike incremental
-    // latency which will be cleared each time dumped. Unit as micro-seconds.
-    //   * @total means io_stats simply adds every fop latency to the counter which is never cleared.
-    //     It is useful for other tools to calculate their metrics report.
-    fop_cumulative_latency_total: [BasicMetric; StatsFop::Max as usize],
-    // Record how many times read latency drops to the ranges.
-    // This helps us to understand the io service time stability.
-    read_latency_dist: [BasicMetric; READ_LATENCY_RANGE_MAX],
-    // Total number of files that are currently open.
-    nr_opens: BasicMetric,
-    // Rwlock closes the race that more than one threads are creating counters concurrently.
-    #[serde(skip_serializing, skip_deserializing)]
-    file_counters: RwLock<HashMap<Inode, Arc<InodeIoStats>>>,
-    #[serde(skip_serializing, skip_deserializing)]
-    access_patterns: RwLock<HashMap<Inode, Arc<AccessPattern>>>,
-    // record regular file read
-    #[serde(skip_serializing, skip_deserializing)]
-    recent_read_files: InodeBitmap,
+/// Trait to manipulate per inode statistics metrics.
+pub trait InodeStatsCounter {
```

Comment:
What benefit does it bring by creating a Trait but only ONE implementation? Or do we have any further plans?

Reply:
Sorry for the noise. It originally had this trait